test/unit/rules/lint-no-negated-condition.js

```diff
@@ -181,6 +176,23 @@ generateRuleTests({
       },
     },
 
+    // Nested inside the body of an `else`:
+    {
+      template: `{{#if condition}}
+                 {{else}}
+                   {{! some comment }}
```

Comment:
Can you add another "bad" test without this comment (I'd like one with the comment and one without it) and without whitespace separating the curlies?

```hbs
{{#if condition}}{{else}}{{#if (not condition)}}<div></div>{{/if}}{{/if}}
```

In this circumstance the resulting AST is _nearly identical_ to:

```hbs
{{#if condition}}{{else if (not condition)}}<div></div>{{/if}}{{/if}}
```

And I want to make sure we are still allowing the latter (since it seems like that was your intent) while disallowing the former...

Reply:
Updated tests as you suggested.